torch/jit/python_frontend.py

```diff
@@ -0,0 +1,285 @@
+        ast.Sub: '-',
+        ast.Mult: '*',
+        ast.Div: '/',
+    }
+
+    @staticmethod
+    def build_Expr(ctx, stmt):
+        return ExprStmt(build_expr(ctx, stmt.value))
+
+    @staticmethod
+    def get_assign_ident(ctx, expr):
+        var = build_expr(ctx, expr)
+        if not isinstance(var, Var):
+            raise RuntimeError("the only expressions allowed on the left hand side of "
+                               "assignments are variable names")
+        return var.name()
+
+    @staticmethod
+    def build_Assign(ctx, stmt):
+        return Assign([StmtBuilder.get_assign_ident(ctx, e) for e in stmt.targets],
+                      '=',
+                      build_expr(ctx, stmt.value))
+
+    @staticmethod
+    def build_AugAssign(ctx, stmt):
+        op = type(stmt.op)
+        if op in StmtBuilder.augassign_map:
+            op_token = StmtBuilder.augassign_map[op]
+        else:
+            raise RuntimeError("unsupported kind of augumented assignment: " + op.__name__)
+        return Assign([StmtBuilder.get_assign_ident(ctx, stmt.target)],
+                      op_token,
+                      build_expr(ctx, stmt.value))
+
+    @staticmethod
+    def build_While(ctx, stmt):
+        if stmt.orelse:
+            raise RuntimeError("else branches of while loops aren't supported")
+        r = ctx.make_range(stmt.lineno, stmt.col_offset, stmt.col_offset + len("while"))
+        return While(r, build_expr(ctx, stmt.test), [build_stmt(ctx, s) for s in stmt.body])
+
+    @staticmethod
+    def build_If(ctx, stmt):
+        r = ctx.make_range(stmt.lineno, stmt.col_offset, stmt.col_offset + len("if"))
+        return If(r, build_expr(ctx, stmt.test),
+                  [build_stmt(ctx, s) for s in stmt.body],
+                  [build_stmt(ctx, s) for s in stmt.orelse])
+
+
+class ExprBuilder(Builder):
+    _MethodRef = namedtuple('MethodRef', ['self', 'name'])
+    binop_map = {
+        ast.Add: partial(BinOp, '+'),
+        ast.Sub: partial(BinOp, '-'),
+        ast.Mult: partial(BinOp, '*'),
+        ast.Div: partial(BinOp, '/'),
+    }
+
+    unop_map = {
+        ast.Not: (UnaryOp, 'not'),
+        ast.USub: (UnaryOp, '-'),
+    }
+
+    boolop_map = {
+        ast.And: partial(BinOp, 'and'),
+        ast.Or: partial(BinOp, 'or'),
+    }
+
+    cmpop_map = {
+        ast.Eq: partial(BinOp, '=='),
+        ast.NotEq: partial(BinOp, '!='),
+        ast.LtE: partial(BinOp, '<='),
+        ast.Lt: partial(BinOp, '<'),
+        ast.GtE: partial(BinOp, '>='),
+        ast.Gt: partial(BinOp, '>'),
+    }
+
+    @staticmethod
+    def build_Attribute(ctx, expr):
+        # NB: the only attributes we support are for getting methods
+        value = build_expr(ctx, expr.value)
+        return ExprBuilder._MethodRef(value, Ident(value.range(), expr.attr))
+
+    @staticmethod
+    def build_Call(ctx, expr):
+        ref = build_expr(ctx, expr.func, allow_methods=True)
+        assert type(ref) is ExprBuilder._MethodRef
```

Comment:
There are no calls supported except methods? The error message should be better...

Reply:
Yes, I put a TODO for error messages at the top and wanted to tackle that as the next change once this lands. Adding nice error handling, with source printing and all that will be +50% code, so I wanted to get this in first, and then convert all ugly asserts and vague "not supported" messages to something better.